rust/src/settings.rs

```diff
@@ -0,0 +1,237 @@
+//! Module for loading and validating coordinator settings.
+//!
+//! Values defined in the configuration file can be overridden by environment variables.
+use crate::mask::{BoundType, DataType, GroupType, MaskConfig, ModelType};
+use config::{Config, ConfigError, Environment};
+use std::path::PathBuf;
+use thiserror::Error;
+use validator::{Validate, ValidationError, ValidationErrors};
+
+#[derive(Error, Debug)]
+pub enum SettingsError {
+    #[error("configuration loading failed: {0}")]
+    Loading(#[from] ConfigError),
+    #[error("validation failed: {0}")]
+    Validation(#[from] ValidationErrors),
+}
+
+#[derive(Debug, Validate, Deserialize, Clone, Copy)]
+#[doc(hidden)]
+pub struct Settings {
+    #[validate]
+    pub api: ApiSettings,
+    #[validate]
+    pub pet: PetSettings,
+    pub mask: MaskSettings,
```

Comment:
i'm not sure why this one is not `#[validate]`d?

Reply:
For `MaskSettings` we don't need to add  `#[validate]` because of the property of `config` for enums. `config` only allows valid enum variances. So `config` already performs the validation. 
If you try to use an invalid value (for example `group_type="None"`), the coordinator will fail with the error: `configuration loading failed: enum GroupType does not have variant constructor None`.